controllers/auth.js

```diff
@@ -0,0 +1,78 @@
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcryptjs')
+const config = require('config')
+
+const User = require('../models/user')
+const validateCreateUser = require('../validations/createUser')
+const validateLoginUser = require('../validations/loginUser')
+
+const createToken = (user) => {
+  const expiresIn = 60 * 60 // one hour
+  const options = { expiresIn }
+  const secret = config.get('jwtPrivateKey')
+
+  const payload = {
+    userId: user._id,
+  }
+
+  return jwt.sign(payload, secret, options)
+}
+
+const responseWithToken = (res, user) => {
+  const token = createToken(user)
+
+  res.header('X-Auth-Token', token).json({
+    error: null,
+    data: {
+      userId: user._id,
+      userEmail: user.email,
+    },
+  })
+}
+
+exports.register = async (req, res) => {
+  const { error } = validateCreateUser(req.body)
+
+  if (error) {
+    return res.status(400).json({ error: error.details[0].message })
```

Comment:
remove json and set send(message) like in others PRs. we need to make it the same in every place to make front work easier

Reply:
every response should be in json format

Reply:
why/

Reply:
we pass only message so on front we will just receive message - this is what we want

Reply:
why? This is not standard, we should work with json and on frontend parse json

Reply:
this is additional work to be done. I checked blogs and some devs send(error), others send(status, {message: error}) and others used json. So this is just the convention and we use send(error) in other places

Reply:
This is no work and what if you want return more data, than simple message?

Reply:
okay leave it with json, probably we will send more properties there

Reply:
but change error object name to message

Reply:
on front we will catch(error) error.message looks better then error.error